hw1/src/Shape.ts

```diff
@@ -0,0 +1,38 @@
+import { Point } from './Point';
+
+export abstract class Shape {
+  protected color: string;
+  protected filled: boolean;
+  protected points: Point[];
+  abstract getType(): string;
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+
+  constructor(points: Point[], color?: string, filled?: boolean) {
+    if(typeof color !== 'string' && typeof color !== 'string') {
+      this.color = 'green';
+      this.filled = true;
+    } else {
+      this.color = color;
+      this.filled = filled;
+    }
+    this.points = points;
+    this.checkNumberOfPoints();
+  }
+
+  private checkNumberOfPoints () {
+    if (!this.points || this.points.length < 3) throw new Error('should fail to be created with 2 points');
+  }
+  
+  public toString() {
```

Comment:
There is no need to define public methods as public explicitly in JavaScript, there are public by default, it's not a mistake though, since it may vary based on the project's code style convention, just FYI